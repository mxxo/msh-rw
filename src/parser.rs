use crate::*;
use std::path::PathBuf;
use nom::*;
use thiserror::Error;

#[derive(Error, Debug)]
pub enum MshError {
    #[error("invalid header (expected {expected:?}, found {found:?})")]
    InvalidHeader {
        expected: String,
        found: String,
    },
}

pub type MshResult<T> = std::result::Result<T, MshError>;

impl Msh2 {
    fn from_file(file: &PathBuf) -> MshResult<Msh2> {
        todo!();
    }
}

#[cfg(test)]
mod tests {
    use super::*;
    #[test]
    fn empty_mesh() {
        let mut path = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
        path.push("props");
        path.push("empty.msh");
        Msh2::from_file(&path).unwrap();
    }
}
