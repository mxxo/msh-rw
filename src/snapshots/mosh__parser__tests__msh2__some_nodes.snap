---
source: src/parser.rs
expression: parse_node_section_msh2(i).unwrap().1
---
[
    Node {
        tag: 1,
        x: 0.0,
        y: 0.0,
        z: 1.0,
    },
    Node {
        tag: 2,
        x: 1.0,
        y: 1.0,
        z: 1.0,
    },
    Node {
        tag: 100,
        x: 1.0,
        y: 1.0,
        z: 1.0,
    },
]
